packages/web/src/pages/upload-page/fields/HiddenAvailabilityFields.tsx

```diff
@@ -0,0 +1,81 @@
+import { Switch } from '@audius/stems'
+import { useField } from 'formik'
+
+import styles from './HiddenAvailabilityFields.module.css'
+import { FIELD_VISIBILITY } from './TrackAvailabilityModalForm'
+
+const messages = {
+  showGenre: 'Show Genre',
+  showMood: 'Show Mood',
+  showTags: 'Show Tags',
+  showShareButton: 'Show Share Button',
+  showPlayCount: 'Show Play Count'
+}
+
+export enum UnlistedTrackMetadataField {
+  UNLISTED = 'unlisted',
+  GENRE = 'genre',
+  MOOD = 'mood',
+  TAGS = 'tags',
+  SHARE = 'share',
+  PLAYS = 'plays'
+}
+
+export const defaultHiddenFields = {
+  genre: true,
+  mood: true,
+  tags: true,
+  share: false,
+  play_count: false
+  // REMIXES handled by a separate field
+}
+
+// The order of toggles in the modal
+const unlistedTrackMetadataOrder = [
+  UnlistedTrackMetadataField.GENRE,
+  UnlistedTrackMetadataField.SHARE,
+  UnlistedTrackMetadataField.MOOD,
+  UnlistedTrackMetadataField.PLAYS,
+  UnlistedTrackMetadataField.TAGS
+]
+
+export const HiddenAvailabilityFields = () => {
+  return (
+    <div className={styles.root}>
+      {unlistedTrackMetadataOrder.map((fieldName) => {
+        return <AvailabilityToggleField key={fieldName} fieldName={fieldName} />
+      })}
+      {/* Dummy row for spacing consistency */}
+      <div className={styles.switchRow}></div>
+    </div>
+  )
+}
+
+const hiddenTrackMetadataMap = {
```

Comment:
Maybe:
```suggestion
const unlistedTrackMetadataMap = {
```
for consistency?

Although a more descriptive name might be something like `messageByFieldName`. Then 76 would look like `messageByFieldName[fieldName]`